addons/web/static/src/js/fields/basic_fields.js

```diff
@@ -1581,6 +1582,442 @@ var FieldBinaryImage = AbstractFieldBinary.extend({
+
+    /**
+     * Renders an empty signature or the saved signature. Both must have the same size.
+     *
+     * @override
+     * @private
+     */
+
+    _render: function () {
+        var signatureRatio = 3;
+        var self = this;
+        var url;
+        var $img;
+        var width = this.nodeOptions.size ? this.nodeOptions.size[0] : this.attrs.width;
+        var height = this.nodeOptions.size ? this.nodeOptions.size[1] : this.attrs.height;
+        if (this.value) {
+            if (!utils.is_bin_size(this.value)) {
+                // Use magic-word technique for detecting image type
+                url = 'data:image/' + (this.file_type_magic_word[this.value[0]] || 'png') + ';base64,' + this.value;
+            } else {
+                url = session.url('/web/image', {
+                    model: this.model,
+                    id: JSON.stringify(this.res_id),
+                    field: this.nodeOptions.preview_image || this.name,
+                    // unique forces a reload of the image when the record has been updated
+                    unique: field_utils.format.datetime(this.recordData.__last_update).replace(/[^0-9]/g, ''),
+                });
+            }
+            $img = $(qweb.render("FieldBinarySignature-img", {widget: this, url: url}));
+        } else {
+            $img = $('<div class="o_signature o_signature_empty"><svg></svg><p>' + _t('SIGNATURE') + '</p></div>');
+            if (width && height) {
+                width = Math.min(width, signatureRatio * height);
+                height = width / signatureRatio;
+            } else if (width) {
+                height = width / signatureRatio;
+            } else if (height) {
+                width = height * signatureRatio;
+            }
+        }
+        if (width) {
+            $img.attr('width', width);
+            $img.css('max-width', width + 'px');
+        }
+        if (height) {
+            $img.attr('height', height);
+            $img.css('max-height', height + 'px');
+        }
+        this.$('> div').remove();
+        this.$('> img').remove();
+
+        this.$el.prepend($img);
+
+        $img.on('error', function () {
+            self._clearFile();
+            $img.attr('src', self.placeholder);
+            self.do_warn(_t("Image"), _t("Could not display the selected image."));
+        });
+    },
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * If the view is in edit mode, open dialog to sign.
+     *
+     * @private
+     */
+    _onClickSignature: function () {
+        if (this.mode === 'edit') {
+            var self = this;
+
+            if (this.nodeOptions.full_name) {
+                var search = "[name='" + this.nodeOptions.full_name + "']";
+                // If full_name correspond to field in read-only mode
+                var signName = $(search).text();
```

Comment:
if you do not store $(search) in a variable, every time you call $(search) there is a lookup on the dom.
I would replace the start of the function by
`var $search = $("[name=.....]");`
then use $search